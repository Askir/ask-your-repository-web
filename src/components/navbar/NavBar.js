// @flow
import React from 'react';
import { connect } from 'react-redux';
import classNames from 'classnames';
import { MdCloudUpload, MdImage } from 'react-icons/md';
import { NavLink } from 'react-router-dom';
import Search from './Search';
import type { Team } from '../../models/Team';
import type { AppState } from '../../state/AppState';
import './NavBar.scss';
import { openTeamSidebar } from '../../state/team_sidebar/teamSidebar.actionCreators';
import TeamInitials from '../team/TeamInitials';

type Props = {
  activeTeam: ?Team,
  isTeamSidebarOpen: boolean,
  onTeamClick: () => void,
};

function NavBar(props: Props) {
  const { isTeamSidebarOpen, onTeamClick, activeTeam } = props;

  return (
    <div className={classNames('NavBar', { 'NavBar--with-sidebar': isTeamSidebarOpen })}>
      <div className="NavBar__left">
        {!isTeamSidebarOpen && activeTeam && (
          <button
            type="button"
            onClick={onTeamClick}
            className="NavBar__team"
          >
            <TeamInitials team={activeTeam} />
          </button>
        )}
        <NavLink
          to="/images"
          className="NavBar__item"
          activeClassName="NavBar__item--active"
        >
          <MdImage className="NavBar__item__icon" />
          <span className="NavBar__item__text">
            Gallery
          </span>
        </NavLink>
      </div>
      <div className="NavBar__search">
        <Search />
      </div>
      <div className="NavBar__right">
        <NavLink
          to="/upload"
          className="NavBar__item"
          activeClassName="NavBar__item--active"
        >
          <MdCloudUpload className="NavBar__item__icon" />
          <span className="NavBar__item__text">
            Upload
          </span>
        </NavLink>
      </div>
    </div>
  );
}

const mapDispatchToProps = (dispatch: Function) => ({
  onTeamClick: () => { dispatch(openTeamSidebar()); },
});

const mapStateToProps = (state: AppState) => ({
  activeTeam: state.activeTeam,
  isTeamSidebarOpen: state.teamSidebar.isOpen,
});

export default connect(mapStateToProps, mapDispatchToProps)(NavBar);
