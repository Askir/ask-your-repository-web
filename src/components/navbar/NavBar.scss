@import '../../style/breakpoints.scss';

.NavBar {
  top: 0;
  left: 0;
  right: 0;
  position: fixed;
  z-index: 100;

  display: flex;
  height: var(--navbar-height);
  background-color: var(--navbar-background-color);

  @media (min-width: $xs-to-sm) {
    padding: 0 var(--padding-sides);
  }

  &__left, &__right {
    display: flex;
    flex: 1;
  }

  &__search { align-self: center; }
  &__right &__item { margin-left: auto; }

  &__item {
    display: flex;
    align-items: center;
    padding: 0 30px;
    font-weight: bold;
    color: white;

    &:hover {
      background-color: var(--accent-color-dark-blue);
    }

    &--active {
      background-color: #111;
      color: var(--accent-color-blue);
    }

    &__icon {
      font-size: 1.5em;
    }

    &__text {
      display: none;
      margin-left: .8rem;

      @media (min-width: $xs-to-sm) {
        display: inline;
      }
    }
  }
}
