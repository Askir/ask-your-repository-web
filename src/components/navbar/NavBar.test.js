// @flow
import React from 'react';
import toJson from 'enzyme-to-json';
import { shallow } from 'enzyme';
import NavBar from './NavBar';

describe('<NavBar />', () => {
  let wrapper;

  beforeEach(() => {
    wrapper = shallow(<NavBar />);
  });

  it('renders correctly', () => {
    expect(toJson(wrapper)).toMatchSnapshot();
  });
});
