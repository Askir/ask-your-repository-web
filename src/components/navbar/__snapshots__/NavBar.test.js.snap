// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<NavBar /> renders correctly 1`] = `
<div
  className="NavBar"
>
  <div
    className="NavBar__left"
  >
    <NavLink
      activeClassName="NavBar__item--active"
      aria-current="page"
      className="NavBar__item"
      to="/images"
    >
      <MdImage
        className="NavBar__item__icon"
      />
      Gallery
    </NavLink>
  </div>
  <div
    className="NavBar__search"
  >
    <Connect(Search) />
  </div>
  <div
    className="NavBar__right"
  >
    <NavLink
      activeClassName="NavBar__item--active"
      aria-current="page"
      className="NavBar__item"
      to="/upload"
    >
      <MdCloudUpload
        className="NavBar__item__icon"
      />
      Upload
    </NavLink>
  </div>
</div>
`;
